### PR TITLE
[lldb] Fix SwiftASTContext creatiion when swift caching is used

### PR DESCRIPTION
**Pull request by cachemeifyoucan**
Followup to #9088, handle the cc1 flag deserialization when swift caching is enabled. LLDB should ignore all the CAS related options from embedded flags since LLDB doesn't receive the correct CAS configuration and should just load the modules from disk directly.

rdar://134090397

### CODE REVIEW

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp [hunk 1]
```diff
@@ -1583,7 +1583,30 @@ void SwiftASTContext::AddExtraClangArgs(const std::vector<std::string> &source,
...
+    });
+
+    clang::DiagnosticsEngine clangDiags(new clang::DiagnosticIDs(),
+                                        new clang::DiagnosticOptions(),
+                                        new clang::IgnoringDiagConsumer());
+    if (!clang::CompilerInvocation::CreateFromArgs(
+          invocation, clangArgs, clangDiags)) {
+      // FIXME: arguments failed to parse so the arguments from source are
+      // invalid. Don't add any arguments for now. Maybe log the error.
+      return;
```
Comment (adrian-prantl) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp]: Can you just call `AddDiagnostic(eSeverityError, message)`?
Hopefully there's a DiagnosticConsumer that can easily convert to a string?

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp [hunk 2]
```diff
@@ -1583,7 +1583,30 @@ void SwiftASTContext::AddExtraClangArgs(const std::vector<std::string> &source,
...
+    }
+
+    // Clear module cache key and other CAS options to load modules from disk
+    // directly.
+    invocation.getFrontendOpts().ModuleCacheKeys.clear();
+    invocation.getCASOpts() = clang::CASOptions();
+
+    invocation.generateCC1CommandLine(
+        [&](const llvm::Twine &arg) { dest.push_back(arg.str()); });
```
Comment (adrian-prantl) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp]: This is really nice & clean!

File: lldb/test/API/lang/swift/clangimporter/caching/TestSwiftClangImporterCaching.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        """
+        self.build()
+        lldbutil.run_to_source_breakpoint(self, "break here",
+                                          lldb.SBFileSpec('main.swift'))
+        log = self.getBuildArtifact("types.log")
+        self.expect('log enable lldb types -f "%s"' % log)
+        self.expect("expression obj", DATA_TYPES_DISPLAYED_CORRECTLY,
+                    substrs=["b ="])
+        self.filecheck('platform shell cat "%s"' % log, __file__)
+#       CHECK:  SwiftASTContextForExpressions(module: "a", cu: "main.swift")::LogConfiguration() --     -cc1
```
Comment (adrian-prantl) [inline lldb/test/API/lang/swift/clangimporter/caching/TestSwiftClangImporterCaching.py]: Should there be a CHECK-NOT for what we filtered out?

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp (hunk 1 above)
Comment (cachemeifyoucan) [reply lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp]: I don't know what is a proper way to hook up (note this is inside a static method). The best way is probably hook up to the clang diagnostics handler and let it emit the error directly maybe? Currently I put in a placeholder ignoreDiagnosticConsumer.

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

**Comment by cachemeifyoucan** [comment]
Please test with following pull request:
https://github.com/swiftlang/swift/pull/75964

@swift-ci please test

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

**Comment by cachemeifyoucan** [comment]
@swift-ci please test macOS platform

**Review by adrian-prantl — APPROVED** [review]

**Comment by adrian-prantl** [comment]
@cachemeifyoucan This is on the wrong branch. Can you rebase this on `swift/release/6.0` and then afterwards cherry-pick it to `stable/202407..` and `next`?

**Comment by cachemeifyoucan** [comment]
> @cachemeifyoucan This is on the wrong branch. Can you rebase this on `swift/release/6.0` and then afterwards cherry-pick it to `stable/202407..` and `next`?

I thought you land on stable/2023... first before gets integrate to release/6.0. Sure I can re-target 6.0 and that should auto forward to stable/2023.

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

**Comment by cachemeifyoucan** [comment]
@swift-ci please test windows platform

**Comment by cachemeifyoucan** [comment]
@adrian-prantl Can you approve and merge for release/6.0?

**Comment by adrian-prantl** [comment]
> I thought you land on stable/2023... first before gets integrate to release/6.0. Sure I can re-target 6.0 and that should auto forward to stable/2023.

The LLDB automerger flows in the opposite direction.

**Pull request merged**